### PR TITLE
update author report to add orcidids

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

For reporting purposes - update the rake task for author export to add orcidid (and alternate identities in csv format)

## How was this change tested?

Running the report in production

### CODE REVIEW

File: lib/tasks/sul.rake [hunk 1]
```diff
@@ -324,41 +324,49 @@ namespace :sul do
     raise 'missing input csv' unless File.file? input_file
     raise 'invalid file format, must be json or csv' unless %w[json csv].include? file_format
 
-    rows = CSV.parse(File.read(input_file), headers: true)
+    rows = CSV.open(input_file, 'rb:bom|utf-8', headers: true).read
     total_authors = rows.size
```
Comment (peetucket) [inline lib/tasks/sul.rake:328]: deal with utf-8 BOM encoding input files

File: lib/tasks/sul.rake [hunk 2]
```diff
@@ -324,41 +324,49 @@ namespace :sul do
...
           author_info = { first_name: author.first_name, middle_name: author.middle_name, last_name: author.last_name,
-                          email: author.emails_for_harvest, sunet: author.sunetid, cap_profile_id: author.cap_profile_id }
-          author_info[:identities] = author.author_identities.map do |ai|
+                          email: author.emails_for_harvest, sunet: author.sunetid, cap_profile_id: author.cap_profile_id, orcid: author.orcidid }
+          author_identities = author.author_identities.map do |ai|
             { first_name: ai.first_name, middle_name: ai.middle_name, last_name: ai.last_name, email: ai.email, institution: ai.institution }
-          end
+          end.compact
+          author_info[:identities] = author_identities unless author_identities.empty?
           output_json_data << author_info
```
Comment (peetucket) [inline lib/tasks/sul.rake:353]: add orcidid to json output, and also ignore adding alternate identities node if its empty

File: lib/tasks/sul.rake [hunk 3]
```diff
@@ -324,41 +324,49 @@ namespace :sul do
...
         else
           CSV.open(output_file, 'a') do |csv|
+            author_identities = author.author_identities.map { |ai| [ai.first_name, ai.middle_name, ai.last_name, ai.email, ai.institution] }.flatten
             csv << [author.first_name, author.middle_name, author.last_name,
-                    author.emails_for_harvest, author.sunetid, author.cap_profile_id]
+                    author.emails_for_harvest, author.sunetid, author.cap_profile_id, author.orcidid, author_identities].flatten
           end
```
Comment (peetucket) [inline lib/tasks/sul.rake:359]: add orcidid and alternate author identities to csv output

File: lib/tasks/sul.rake [hunk 4]
```diff
@@ -324,41 +324,49 @@ namespace :sul do
...
+        found += 1
       else
         puts "***** ERROR: Sunet: #{sunet}, cap_profile_id: #{cap_profile_id} not found, skipping"
+        skipped += 1
         CSV.open(output_file, 'a') { |csv| csv << ['', '', '', '', sunet, cap_profile_id] } if file_format == 'csv'
       end
     end
     File.write(output_file, JSON.pretty_generate(output_json_data)) if file_format == 'json'
+    puts "Input rows: #{total_authors}, found: #{found}, not found: #{skipped}.  Results written to #{output_file}."
   end
```
Comment (peetucket) [inline lib/tasks/sul.rake:370]: add some stats for output when done

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**